os/drivers/audio/alc5658.c

```diff
@@ -673,16 +713,16 @@ static int alc5658_start(FAR struct audio_lowerhalf_s *dev)
 {
 
 	FAR struct alc5658_dev_s *priv = (FAR struct alc5658_dev_s *)dev;
-
+	t_codec_init_script_entry *sample;
 	if (priv->running) {
 		return OK;
 	}
 
 	audvdbg(" alc5658_start Entry\n");
 	alc5658_exec_i2c_script(priv, codec_init_inout_script1, sizeof(codec_init_inout_script1) / sizeof(t_codec_init_script_entry));
 
-	/* Fix me -- Need to support multiple samplerate */
-	alc5658_exec_i2c_script(priv, codec_init_pll_16K, sizeof(codec_init_pll_16K) / sizeof(t_codec_init_script_entry));
+	sample = alc5658_get_sample_rate_script(priv->samprate);
+	alc5658_exec_i2c_script(priv, sample, sizeof(sample) / sizeof(t_codec_init_script_entry));
```

Comment:
We should a NULL checker here, if an unsupported sampling rate is specified then an error should be thrown.

Reply:
I added NULL checking in configuration called from pcm_set_config, here probably will be called from pcm_start, so no more NULL checking needed.
But to prevent rare case, I will apply your opinion.